tests/tests/capsule_test.cpp

```diff
@@ -0,0 +1,118 @@
+
+#include <boost/test/unit_test.hpp>
+
+#include <koinos/capsule/capsule.hpp>
+#include <koinos/tests/capsule/dummy_hash.hpp>
+#include <koinos/tests/capsule/dummy_operations.hpp>
+#include <koinos/tests/capsule/dummy_tree.hpp>
+
+#include <memory>
+#include <string>
+#include <tuple>
+#include <variant>
+
+struct capsule_fixture
+{
+   capsule_fixture()
+   {
+   }
+
+   ~capsule_fixture()
+   {
+   }
+};
+
+BOOST_FIXTURE_TEST_SUITE( capsule_tests, capsule_fixture )
+
+using namespace koinos::capsule;
+
+BOOST_AUTO_TEST_CASE( dummy_hash_test )
+{ try {
+   BOOST_TEST_MESSAGE( "Checking dummy hash functionality" );
+   auto h25 = create_dummy_hash(25);
+   auto h29 = create_dummy_hash(29);
+   auto h37 = create_dummy_hash(37);
+   auto ha = reduce_dummy_hash(h25, h29);
+   auto hb = reduce_dummy_hash(ha, h37);
+   BOOST_CHECK_EQUAL( dummy_hash_to_string(hb), "((h25, h29), h37)" );
+} KOINOS_CATCH_LOG_AND_RETHROW(info) }
+
+BOOST_AUTO_TEST_CASE( tree_walker_test_table_cases )
+{ try {
+   BOOST_TEST_MESSAGE( "Tree walker functionality" );
+
+   //                                15
+   //                         /               \
+   //                 7                              23
+   //             /       \                       /      \
+   //         3              11              19              27
+   //       /   \          /    \          /    \          /    \
+   //     1       5       9      13      17      21      25      29
+   //    / \     / \     / \     / \     / \     / \     / \     / \
+   //   0   2   4   6   8  10  12   14 16   18 20   22 24   26 28   30
+
+   std::vector< std::tuple< int64_t, std::vector< int64_t >, std::string > > cases = {
```

Comment:
We should avoid using nested vectors of magic numbers, it makes the test cases hard to understand and maintain.